### ISSUE TITLE
PeriodicUpdater related segfault

### ISSUE THREAD

**Issue by filimonov** [yandex/ClickHouse#6607]
last logged actions from thread 72:
```
2019.08.21 16:37:12.523680 [ 72 ] {} <Warning> ConnectionPoolWithFailover: Connection failed at try №1, reason: Code: 210, e.displayText() = DB::NetException: Connection refused (data-dev-we-ch-sh1r2.host:9440) (version 19.11.3.11)
2019.08.21 16:37:12.524304 [ 72 ] {} <Warning> ConnectionPoolWithFailover: Connection failed at try №2, reason: Code: 210, e.displayText() = DB::NetException: Connection refused (data-dev-we-ch-sh1r2.host:9440) (version 19.11.3.11)
2019.08.21 16:37:12.525117 [ 72 ] {} <Warning> ConnectionPoolWithFailover: Connection failed at try №3, reason: Code: 210, e.displayText() = DB::NetException: Connection refused (data-dev-we-ch-sh1r2.host:9440) (version 19.11.3.11)
```

```
2019.08.21 16:38:33.342326 [ 102 ] {} <Error> BaseDaemon: ########################################
2019.08.21 16:38:33.342405 [ 102 ] {} <Error> BaseDaemon: (version 19.11.3.11) (from thread 72) Received signal Segmentation fault (11).
2019.08.21 16:38:33.342467 [ 102 ] {} <Error> BaseDaemon: Address: NULL pointer. Access: read. Address not mapped to object.
2019.08.21 16:38:33.362219 [ 102 ] {} <Error> BaseDaemon: 0. clickhouse-server(StackTrace::StackTrace(ucontext const&)+0x22) [0x6d0af52]
1. clickhouse-server() [0x31caa35]
2. /lib64/libpthread.so.0(+0xf5d0) [0x7f2dc10e25d0]
3. clickhouse-server(std::_Function_handler<bool (std::shared_ptr<DB::IExternalLoadable const> const&), std::_Bind<bool (DB::IExternalLoadable::*(std::_Placeholder<1>))() const> >::_M_invoke(std::_Any_data const&, std::shared_ptr<DB::IExternalLoadable const> const&)+0x15) [0x5c7e105]
4. clickhouse-server(DB::ExternalLoader::LoadingDispatcher::reloadOutdated()+0x17a) [0x5c85a2a]
5. clickhouse-server(DB::ExternalLoader::PeriodicUpdater::doPeriodicUpdates()+0x90) [0x5c85f00]
6. clickhouse-server(ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::ExternalLoader::PeriodicUpdater::*)(), DB::ExternalLoader::PeriodicUpdater*>(void (DB::ExternalLoader::PeriodicUpdater::*&&)(), DB::ExternalLoader::PeriodicUpdater*&&)::{lambda()#1}::operator()() const+0x35) [0x5c7e7a5]
7. clickhouse-server(ThreadPoolImpl<std::thread>::worker(std::_List_iterator<std::thread>)+0x1b8) [0x6641888]
8. clickhouse-server() [0x74ad71f]
9. /lib64/libpthread.so.0(+0x7dd5) [0x7f2dc10dadd5]
10. /lib64/libc.so.6(clone+0x6d) [0x7f2dc0b01ead]
```

**Comment by alexey-milovidov** [comment]
@vitlibar Please confirm that it is fixed in recent versions.

**Comment by vitlibar** [comment]
Fixed in https://github.com/yandex/ClickHouse/pull/6082
The fix was in the release starting from 19.11.4.24.
